uamqp/utils.py

```diff
@@ -93,7 +94,44 @@ def add_batch(batch, message):
     batch.data.append(output)
 
 
+def encode_str(data, encoding='utf-8'):
+    try:
+        return data.encode(encoding)
+    except AttributeError:
+        return data
+
+
+def normalized_data_body(data, **kwargs):
+    encoding = kwargs.get("encoding", "utf-8")
+    if isinstance(data, list):
+        return [encode_str(item, encoding) for item in data]
+    else:
+        return [encode_str(data, encoding)]
+
+
+def normalized_sequence_body(sequence):
+    if isinstance(sequence, list) and all([isinstance(b, list) for b in sequence]):
+        return sequence
+    elif isinstance(sequence, list):
+        return [sequence]
+
+
 def get_message_encoded_size(message):
     output = bytearray()
     encode_payload(output, message)
     return len(output)
+
+
+# TODO: to have separate module/class to host those helper AMQP VALUE wrapper method
```

Comment:
Wrapping things like this in methods is expensive. Method calls in Python are not performant. So we'll need to look at the balance between readable/maintainable code - but also not putting extra method calls along the critical perf path.

Reply:
there could be cases that the AMQP spec is explicitly asking for type info like ulong/uint/long/int if the value is an integer.

however, python only have int. so we need to pass extra type information for the correct encoding of an integer to tell whether it's a AMQP long, AMQP short, AMQP int etc.
I think for most cases, default Python-AMQP type mapping is good, but we need a way to override the default type.
right now those utils methods are called by the eventhub sdk.